src/sage/rings/ideal.py

```diff
@@ -622,7 +622,7 @@ def reduce(self, f):
         """
         return f       # default
 
-    def gens(self):
+    def gens(self):  # -> tuple | PolynomialSequence
```

Comment:
And here

Reply:
also not a class

Reply:
and I did not make any serious analysis of the possible types that can appear here

Reply:
This could use `from sage.structure.sequence import Sequence_generic` maybe. But I would rather keep it as it is now.